adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralNativeRenderer.m

```diff
@@ -0,0 +1,252 @@
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationNativeLoadCompletionHandler originalCompletionHandler =
+        [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationNativeAdEventDelegate>(
+        _Nullable id<GADMediationNativeAd> ad, NSError *_Nullable error) {
+      if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+        return nil;
+      }
+      id<GADMediationNativeAdEventDelegate> delegate = nil;
+      if (originalCompletionHandler) {
+        delegate = originalCompletionHandler(ad, error);
+      }
+      originalCompletionHandler = nil;
+      return delegate;
+    };
+    
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    UIViewController *rootViewController = adConfiguration.topViewController;
+    
+    if ([GADMAdapterMintegralUtils isStringEmpty:adUnitId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+        
+    _adUnitId = adUnitId;
+    _nativeManager = [[MTGBidNativeAdManager alloc]initWithPlacementId:placementId unitID:adUnitId presentingViewController:rootViewController];
+    _nativeManager.delegate = self;
+    [_nativeManager loadWithBidToken:adConfiguration.bidResponse];
+}
+
+- (MTGMediaView *)createMediaView {
+    if (_mediaView) {
+        return _mediaView;
+    }
+    _mediaView= [[MTGMediaView alloc] initWithFrame:CGRectZero];
+    _mediaView.delegate = self;
+    return _mediaView;
+}
+
+#pragma mark - MTGBidNativeAdManagerDelegate
+- (void)nativeAdsLoaded:(nullable NSArray *)nativeAds bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    
+    if ([nativeAds isKindOfClass:NSArray.class] && nativeAds.count > 0) {
+        _campaign = nativeAds.firstObject;
+        
+        if (![GADMAdapterMintegralUtils isStringEmpty:_campaign.iconUrl]) {
+            _icon = [self loadImageWithURLString:_campaign.iconUrl];
+        }
+                
+        if (![GADMAdapterMintegralUtils isStringEmpty:_campaign.imageUrl]) {
+            GADNativeAdImage *image = [self loadImageWithURLString:_campaign.imageUrl];
+            if (image) {
+                _images = @[image];
+            }
+        }
+        
+        [self createMediaView];
+        
+        if (_adLoadCompletionHandler) {
+            _adEventDelegate = _adLoadCompletionHandler(self,nil);
+        }
+
+    }else{
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorAdNotFill, @"Native ad not fill");
```

Comment:
Update the error message "No native ad returned from Mintegral."

Reply:
ok, i have changed.